gc/base/MemoryPoolAddressOrderedListBase.cpp

```diff
@@ -168,3 +168,34 @@ MM_MemoryPoolAddressOrderedListBase::abandonMemoryInPool(MM_EnvironmentBase* env
 	abandonHeapChunk((MM_HeapLinkedFreeHeader*)address, (uint8_t*)address + size);
 }
 
+#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+uintptr_t
+MM_MemoryPoolAddressOrderedListBase::releaseMemoryPoolFreeEntryMemoryPages(MM_EnvironmentBase* env, MM_HeapLinkedFreeHeader* freeEntry)
+{
+	uintptr_t freedMemory = 0;
+	MM_HeapLinkedFreeHeader* currentFreeEntry = freeEntry;
+	uintptr_t pageSize = env->getExtensions()->heap->getPageSize();
+	while (NULL != currentFreeEntry) {
+		if (currentFreeEntry->getSize() >= ((2 * pageSize) + sizeof(MM_HeapLinkedFreeHeader))) {
+			/* release/decommit memory after Header */
+			uint8_t* addressBase = (uint8_t*)MM_Math::roundToCeiling(pageSize, (uintptr_t)((uint8_t*)currentFreeEntry + sizeof(MM_HeapLinkedFreeHeader)));
```

Comment:
Seems to me that this line is redunant:
if (currentFreeEntry->getSize() >= ((2 * pageSize) + sizeof(MM_HeapLinkedFreeHeader))) {

it will be later verified with totalFreePagesCount.